enderio-base/src/main/java/com/enderio/base/common/loot/BrokenSpawnerLootModifier.java

```diff
@@ -41,11 +40,16 @@ protected BrokenSpawnerLootModifier(LootItemCondition[] conditionsIn) {
     protected List<ItemStack> doApply(List<ItemStack> generatedLoot, LootContext context) {
         BlockEntity entity = context.getParam(LootContextParams.BLOCK_ENTITY);
         if (entity instanceof SpawnerBlockEntity spawnerBlockEntity) {
-            if (context.getParam(LootContextParams.TOOL).is(EIOTags.Items.BROKEN_SPAWNER_BLACKLIST)) {
+            if (!context.getParam(LootContextParams.TOOL).is(EIOTags.Items.BROKEN_SPAWNER_BLACKLIST)) {
                 if (context.getRandom().nextFloat() < BaseConfig.COMMON.BLOCKS.BROKEN_SPAWNER_DROP_CHANCE.get()) {
                     BaseSpawner spawner = spawnerBlockEntity.getSpawner();
-                    ItemStack brokenSpawner = BrokenSpawnerItem.forType(spawner.getEntityId(context.getLevel(), entity.getBlockPos()));
-                    return Lists.newArrayList(brokenSpawner);
+                    CompoundTag entityTag = spawner.nextSpawnData.getEntityToSpawn();
+
+                    if (entityTag.contains("id")) {
+                        ResourceLocation type = new ResourceLocation(entityTag.getString("id"));
+                        ItemStack brokenSpawner = BrokenSpawnerItem.forType(type);
+                        return Lists.newArrayList(brokenSpawner);

```

Comment:
maybe append to the list instead of returning a list only containing the broken spawner